### PR TITLE
Add resources to AlertManager

### PR DESCRIPTION
**Pull request by latchmihay**
Adding default AlertManager resources

### CODE REVIEW

**Comment by metalmatze** [comment]
You need to run `make generate` once and commit the changes. Thanks!

File: jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet [hunk 1]
```diff
@@ -40,6 +40,10 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         ],
       },
       replicas: 3,
+      resources: {
+        imits: { "cpu": "100m",  "memory": "100Mi" },
```
Comment (BenoitKnecht) [inline jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: Small typo here, it should be `limits`.

Also, to be consistent with the rest of the file, the `limits` and `requests` objects should be formatted as

```jsonnet
{ cpu: '100m', memory: '100Mi' }
```

**Comment by BenoitKnecht** [comment]
@latchmihay Other than that, I think you can just drop 94cce416474a3b0f1a92d137abffef48158174f5; it will resolve the merge conflict, and as far as I can tell, your change doesn't require a dependency update.

File: jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet (hunk 1 above)
Comment (latchmihay) [reply jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: thank you for your review, I have fixed it ;)

**Comment by BenoitKnecht** [comment]
@latchmihay You're almost there... You just need to rebase on master, and drop the changes to `jsonnetfile.lock.json`.

File: jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet [hunk 2]
```diff
@@ -40,6 +40,10 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         ],
       },
       replicas: 3,
+      resources: {
+        limits: { cpu: '100m', memory: '100Mi' },
+        requests: { cpu: '100m', memory: '100Mi' },
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: Why limits and requests are the same?

Comment (metalmatze) [reply jsonnet/kube-prometheus/alertmanager/alertmanager.libsonnet]: I would guess to make sure that we can schedule and always have these resources available.

**Comment by BenoitKnecht** [comment]
@latchmihay Maybe try something like this:

```console
$ git checkout alertManagerResouces
$ git fetch origin
$ git rebase -i origin/master
```

In the editor that opens, keep the following lines only:

```
pick 88e10af - adding resources to alertmanager
edit c73297e fixing a small typo in limits and adding standard format to resources, and updating lock file to latest from upstream
pick 6a175ec regenerating yamls
```

Save and exit the editor, then run

```console
$ git reset HEAD^ -- jsonnetfile.lock.json
$ git commit --amend
```

Edit the comment, save and exit the editor, and then

```console
$ git checkout jsonnetfile.lock.json
$ git rebase --continue
```

At this point, `git log origin/master..` should only show three commits, and `git diff origin/master` should only show a few added lines.

Now you can `git push -f` to update this PR.

**Comment by latchmihay** [comment]
Thank you for the set of instructions. They worked nicely. @BenoitKnecht

**Comment by BenoitKnecht** [comment]
@latchmihay This time I'm pretty sure it's just bad luck and the pipeline failure is unrelated to your change.

@metalmatze @paulfantom I don't know which slash command can be used to rerun the pipeline (and I don't know if regular users are even allowed to do that).

**Comment by brancz** [comment]
Re-running. Let's see if it's just a flake.

**Comment by BenoitKnecht** [comment]
So the failure seems consistent: `main_test.go:95: wait for prometheus-k8s: timed out waiting for the condition`.

@brancz Is it possible that the resource reservation added in this PR are causing the k8s scheduler to fail to start the Prometheus pods? I don't know how large the CI k8s cluster is, but given that the Alertmanager is replicated 3 times, maybe it's pushing it over the limit.

**Comment by paulfantom** [comment]
Since we refactored codebase, this is no longer valid. I am moving it to #984. Sorry it took so much time to see progress on this.

**Pull request closed without merging**